Lib/gftools/packager.py

```diff
@@ -0,0 +1,1312 @@
+                        Enum,
+                        Seq,
+                        Str,
+                        Any,
+                        EmptyNone,
+                        EmptyDict,
+                        dirty_load,
+                        as_document,
+                        YAMLValidationError,
+                        YAML
+                      )
+
+from warnings import warn
+import functools
+from google.protobuf import text_format
+import gftools.fonts_public_pb2 as fonts_pb2
+
+
+GITHUB_REPO_HTTPS_URL = 'https://github.com/{gh_repo_name_with_owner}.git'.format
+GITHUB_REPO_SSH_URL = 'git@github.com:{gh_repo_name_with_owner}.git'.format
+
+GITHUB_GRAPHQL_API = 'https://api.github.com/graphql'
+GITHUB_V3_REST_API = 'https://api.github.com/'
+
+# Using object(expression:$rev), we query all three license folders
+# for family_name, but only the entry that exists will return a tree (directory).
+# Non existing directories will be null (i.e. None).
+# Hence, we can send one query to know the family exists or not, in which
+# directory (license) and have a (flat) directory listing.
+# I queried "rateLimit{cost}" and this had a cost of 1!
+GITHUB_GRAPHQL_GET_FAMILY_ENTRY = """
+fragment FamilyFiles on Tree {
+  entries{
+    name
+    type
+    oid
+  }
+}
+
+query ListFiles($repoName: String!,
+                $repoOwner: String!,
+                $reference: String!,
+                $oflDir: String!,
+                $uflDir: String!,
+                $apacheDir: String!
+) {
+  repository(name: $repoName, owner: $repoOwner) {
+    ref(qualifiedName: $reference) {
+      prefix
+      name
+      target {
+        ... on Commit {
+          __typename
+          oid
+          messageHeadline
+          pushedDate
+        }
+      }
+    }
+    ofl: object(expression: $oflDir) {
+      ...FamilyFiles
+    }
+    apache: object(expression:$apacheDir) {
+      __typename
+      ...FamilyFiles
+    }
+    ufl: object(expression:$uflDir) {
+      __typename
+      ...FamilyFiles
+    }
+  }
+}
+"""
+
+def _get_query_variables(repo_owner, repo_name, family_name, reference='refs/heads/master'):
+  """
+  call like: get_query_variables('google', 'fonts', 'gelasio')
+
+  reference: see $ git help rev-parse
+            and git help revisions
+            and https://git-scm.com/book/en/v2/Git-Internals-Git-References
+  for a branch called "master" "refs/heads/master" is best, but
+  "master" would work as well.
+  tag names work as well, ideally "ref/tags/v0.6.8" but "v0.6.8" would
+  work too. The full name is less ambiguous.
+  """
+  return {
+    'repoOwner': repo_owner,
+    'repoName': repo_name,
+    'reference': reference,
+    'oflDir': f'{reference}:ofl/{family_name}',
+    'apacheDir': f'{reference}:apache/{family_name}',
+    'uflDir': f'{reference}:ufl/{family_name}'
+  }
+
+def _get_github_api_token() -> str:
+  # $ export GITHUB_API_TOKEN={the GitHub API token}
+  return os.environ['GITHUB_API_TOKEN']
```

Comment:
Would you mind calling this `GH_TOKEN`?

I'm already using it here, https://github.com/googlefonts/gftools/blob/master/bin/gftools-qa.py#L327

Reply:
No problem, will do.